server/src/com/cloud/network/IpAddressManagerImpl.java

```diff
@@ -901,40 +916,28 @@ public PublicIp assignDedicateIpAddress(Account owner, final Long guestNtwkId, f
 
         PublicIp ip = null;
         try {
-            ip = Transaction.execute(new TransactionCallbackWithException<PublicIp, InsufficientAddressCapacityException>() {
```

Comment:
@yvsubhash this method is no longer bound in transaction and may cause failures.

Reply:
@yvsubhash I've found a regression with changes to this method. It causes SNAT IPs to not get allocated and we see something like this:
```
2017-12-20 14:32:40,402 ERROR [c.c.n.IpAddressManagerImpl] (API-Job-Executor-16:ctx-d4229b63 job-418 ctx-da9f0f88) (logid:8f31a2f6) Unable to get source nat ip address for account 2 
```
This causes VPC/VPN tests to fail.

Reply:
@yvsubhash maybe re-introduce a transaction? Maybe without a locking/transaction multiple parties may receive the same public IP.